halo2/src/prover.rs

```diff
@@ -205,28 +324,93 @@ impl<'a, F: FieldElement> Halo2Prover<'a, F> {
         }
 
         log::info!("Verifying aggregated proof in the EVM...");
-        aggregation::evm_verify(deployment_code, agg_circuit_with_proof.instances(), &proof);
+        evm_verify(deployment_code, agg_instances.clone(), &proof);
+
+        let output_dir = self.output_dir.ok_or("output_dir is None".to_owned())?;
```

Comment:
I think we can handle `output_dir` of `None`?

Reply:
What do we use then?

Reply:
We don't write the publics. That's the only reason it's used, right?

Actually, I think the prover should not know about the output dir at all, and instead return the publics together with the proof. And then it is up to the pipeline to persist this or not.

Reply:
To me it would be a cleaner separation if the "backend" is only responsible for computing proofs and publics and returns both. You should be able to compute proofs without ever touching the disk. In the case of the Polygon eStark backend it might have to persist files, because it needs to invoke the prover in a different process and it expects file on disk. But that's an implementation detail of this specific backend.